tests/cpp/test_multidevice_tutorial.cpp

```diff
@@ -615,4 +615,509 @@ TEST_F(MultiDeviceTutorial, TensorShardingAndResharding) {
   }
 }
 
+// We have seen how to multidevice-schedule a Fusion by setting tensors'
+// sharding, i.e., setting device mesh and setting axis parallel type to DIDx.
+
+// At the time where this comment is written, we only support 1D tensor
+// sharding, i.e., we have not introduced yet multidimensional meshes and other
+// parallel types such as DIDy and DIDz. This will be done in the future. Let us
+// poit out by the way that 1D multidevice parallelism already covers
+// interesting real-world scenarios, such as Transformer with Tensor Parallel,
+// see `tests/cpp/test_multidevice_transformer.cpp`.
+
+// Let us now explore lower than the mere multidevice scheduling API and
+// introduce the host IR. In classical single-device nvFuser, the "host program"
+// could be summarized as launching one or a couple of CUDA Kernels. When
+// dealing with multiple devices, the host plays a more proeminent role because
+// it needs to orchestrate and synchronize compute Kernels and Communications
+// (which are necessarily CPU-initiated when using NCCL or UCC). Other examples,
+// not necessarily tied to multidevice setting, rely on complex host
+// orchestration, such as multi streaming, kernel pipelining, overlap technics,
+// using CUDA Graphs, etc.
+
+// These examples suggest that Fuser should be able to reason about the
+// interplay between host and device execution. This motivates the introduction
+// of "host IRs" to represent the host program, in an abstract/symbolic way,
+// allowing us reason about the host program, apply optimizations passes, and
+// possibly compile it (in the future).
+
+// The HostIr component is comprised of three parts:
+// - Host IRs: each IR represents an elementary host operation
+// - HostIrContainer: represents the host program
+// - HostIrExecutor: executes a HostIrContainer
+
+// The host program is typically generated automatically during lowering. This
+// is what is done at the instantiation of MultiDeviceExecutor, and what gets
+// printed by `MultiDeviceExecutor::print()`. However, the Host Ir API has been
+// designed to allow a fully manual host programmation. This is what we are
+// going to introduce in the following tests.
+
+//  Let us start with the simplest non-trivial host program possible: compiling
+//  and running a single Fusion. It is a good starting point to understand the
+//  Host Ir semantics. The host program could be illustrated as follows:
+/*
+  | tv0: input
+  | tv1 = Fusion (tv0)
+  | tv1: output
+*/
+
+namespace hir { // HostIr has its own namespace "hir"
+
+TEST_F(MultiDeviceTutorial, HostIrLaunchingFusion) {
+  // Let us consider an arbitrary fusion. We assume for simplicity (but without
+  // loss of generality) that the fusion has one input and one output which are
+  // both a 2D tensor.
+  constexpr int64_t kNDims = 2;
+  auto CreateFusion = []() -> std::unique_ptr<Fusion> {
```

Comment:
Naming style:
```suggestion
  auto create_fusion = []() -> std::unique_ptr<Fusion> {
```
or `createFusion`. Functors can be treated as variables or functions, so either is fine. The former is more commonly seen in nvFuser's code base.